perf_benchmark/benchmark_input_stream_test.cc

```diff
@@ -0,0 +1,288 @@
+// Copyright 2016 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+//
+#include "gtest/gtest.h"
+#include "perf_benchmark/benchmark_input_stream.h"
+#include "perf_benchmark/utils.h"
+#include "grpc_transcoding/json_request_translator.h"
+#include "grpc_transcoding/type_helper.h"
+#include "perf_benchmark/benchmark.pb.h"
+#include "google/api/service.pb.h"
+#include "absl/strings/str_format.h"
+#include "absl/strings/escaping.h"
+
+namespace google {
+namespace grpc {
+namespace transcoding {
+
+namespace perf_benchmark {
+namespace {
+constexpr absl::string_view
+    kServiceConfigTextProtoFile = "benchmark_service.textproto";
+
+// Global type helper containing the type information of the benchmark_service
+// service config object.
+[[nodiscard]] const TypeHelper& GetBenchmarkTypeHelper() {
+  static const auto* const kTypeHelper = [] {
+    // Load service config proto into Service object.
+    // Construct object on the heap using new without calling its dtor to
+    // avoid destruction issue with static variables. However, this can cause
+    // unnecessary heap allocations and create minor performance concerns.
+    // For a small benchmark script, this is okay.
+    auto* service = new google::api::Service();
+    GOOGLE_CHECK_OK(
+        LoadService(std::string(kServiceConfigTextProtoFile), service));
+
+    // Create a TypeHelper based on the service config.
+    // Construct object on the heap for the same reason as the Service config.
+    auto* type_helper = new TypeHelper(service->types(), service->enums());
+    return type_helper;
+  }();
+  return *kTypeHelper;
+}
+
+std::string ParseJsonMessageToProtoMessage(absl::string_view json_msg,
```

Comment:
hmm,  this is mis-leading.  By reading the file name, I thought it is a unit-test for input_stream test,  but it is more that.  Could you split it into a sparate test.cc file?